packages/components/src/test/properties/NavigationPropertyTargetSelector.test.tsx

```diff
@@ -168,16 +160,153 @@ describe("NavigationPropertyTargetSelector", () => {
+    const value = {
+      valueFormat: PropertyValueFormat.Primitive,
+      value: { id: "1", className: "TestSchema:TargetClass" },
+      displayValue: "Target Class Instance",
+    };
+    const propertyRecord = new PropertyRecord(value as PropertyValue, propertyDescription);
+
+    const initialProps: NavigationPropertyTargetSelectorProps = {
+      imodel: testImodel,
+      getNavigationPropertyInfo: async () => testNavigationPropertyInfo,
+      propertyRecord,
+    };
+    const { container, queryByText } = render(<NavigationPropertyTargetSelector {...initialProps} />);
+
+    const dropdownButton = await waitFor(() => {
+      const element = container.querySelector<HTMLDivElement>(".iui-end-icon");
+      expect(element).to.not.be.null;
+      return element;
+    });
+
+    fireEvent.click(dropdownButton!);
+
+    await waitFor(() => expect(queryByText(contentItem.label.displayValue)).to.not.be.null);
+
+    fireEvent.click(dropdownButton!);
+
+    await waitFor(() => expect(queryByText(contentItem.label.displayValue)).to.be.null);
+  });
+
+  it("handles input blur", async () => {
+    sinon.stub(Presentation.presentation, "getContent").resolves(new Content(createTestContentDescriptor({ fields: [], categories: [] }), [contentItem]));
+    const propertyDescription = createNavigationPropertyDescription();
+    const value = {
+      valueFormat: PropertyValueFormat.Primitive,
+      value: { id: "1", className: "TestSchema:TargetClass" },
+      displayValue: "",
+    };
+    const propertyRecord = new PropertyRecord(value as PropertyValue, propertyDescription);
+
+    const initialProps: NavigationPropertyTargetSelectorProps = {
+      imodel: testImodel,
+      getNavigationPropertyInfo: async () => testNavigationPropertyInfo,
+      propertyRecord,
+    };
+    const { container, getByDisplayValue, getByRole, queryByText } = render(<NavigationPropertyTargetSelector {...initialProps} />);
+
+    const dropdownButton = await waitFor(() => {
+      const element = container.querySelector<HTMLDivElement>(".iui-end-icon");
+      expect(element).to.not.be.null;
+      return element;
+    });
+
+    // when input value is empty
+    fireEvent.click(dropdownButton!);
+    await waitFor(() => expect(queryByText(contentItem.label.displayValue)).to.not.be.null);
+    fireEvent.click(dropdownButton!);
+    await waitFor(() => expect(queryByText(contentItem.label.displayValue)).to.be.null);
+    expect((getByRole("textbox") as HTMLInputElement).value).to.be.eq("");
+
+    // when input value is not empty
+    fireEvent.click(dropdownButton!);
+    await waitFor(() => expect(queryByText(contentItem.label.displayValue)).to.not.be.null);
+    const menuItem = queryByText(contentItem.label.displayValue);
+    fireEvent.click(menuItem!);
+    await waitFor(() => expect(queryByText(contentItem.label.displayValue)).to.be.null);
+    expect(getByDisplayValue(contentItem.label.displayValue)).to.not.be.null;
```

Comment:
I think it would be more clear what you are checking if you made this check same as when input is empty:
```
expect((getByRole("textbox") as HTMLInputElement).value).to.be.eq(contentItem.label.displayValue)
```

Reply:
fixed